app/javascript/components/visualization/ScatterPlot.js

```diff
@@ -119,7 +119,7 @@ function RawScatterPlot({
       })
       store.addNotification(successNotification(`Colors saved successfully`))
       const newScatterData = Object.assign({}, scatterData, {
-        customColors: response.cluster_file_info.custom_colors[annotation.name]
+        customColors: response.cluster_file_info?.custom_colors[scatterData.annotParams.name] ?? {}
```

Comment:
I'm not sure i've seen a double question mark operator before - is this essentially a check that the first part has something and if not assign it to be an empty object?

Reply:
Yup, I just learned about it last week.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator